src/main/java/net/sf/jabref/model/entry/EntryType.java

```diff
@@ -52,10 +50,35 @@
     /**
      * TODO: move inside GUI
      */
-    List<String> getPrimaryOptionalFields();
+    public abstract List<String> getPrimaryOptionalFields();
 
     /**
      * TODO: move inside GUI
      */
-    List<String> getSecondaryOptionalFields();
+    public abstract List<String> getSecondaryOptionalFields();
+
+    @Override
+    public boolean equals(Object other) {
+        if (other == null) {
+            return false;
+        }
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof EntryType)) {
+            return false;
+        }
+        EntryType otherEntryType = (EntryType) other;
+        return otherEntryType.getName().equals(this.getName());
+    }
+
+    @Override
+    public int hashCode() {
+        return this.getName().hashCode();
+    }
+
+    @Override
+    public int compareTo(EntryType o) {
+        return getName().compareTo(o.getName());
+    }
```

Comment:
I think `Comparator` is not applicable here. What is actually required is a proper equals method, the comparison is just an extra.